corehq/util/pagination.py

```diff
@@ -182,6 +195,31 @@ def _get_event_handler(self):
         else:
             return ResumableIteratorEventHandler(self)
 
+    def retry(self, item_id, max_retry=3):
+        """Add document to be yielded at end of iteration
+
+        Iteration order of retry documents is undefined. All retry
+        documents will be yielded after the initial non-retry phase of
+        iteration has completed, and every retry document will be
+        yielded each time the iterator is stopped and resumed during the
+        retry phase. This method is relatively inefficient because it
+        forces the iteration state to be saved to couch. If you find
```

Comment:
No specific use case in mind right now, but multiple databases can exist having the same doc types. It's easier to reason about data consistency if the docs and the iteration state live in the same db.